virttest/test_setup.py

```diff
@@ -298,17 +288,13 @@ def check_status_with_value(action_list, file_name):
             for (act, ret) in action_list:
                 logging.info("Writing path %s: %s, expected khugepage rc: %s ",
                              file_name, act, ret)
-                try:
-                    file_object = open(file_name, "w")
-                    file_object.write(act)
-                    file_object.close()
-                except IOError as error_detail:
-                    logging.info("IO Operation on path %s failed: %s",
-                                 file_name, error_detail)
+                khugepage = kernel_interface.SysFS(file_name, session=self.session)
+                khugepage.sys_fs_value = act
                 timeout = time.time() + 50
                 while time.time() < timeout:
                     try:
-                        process.run('pgrep khugepaged', verbose=False)
+                        ret = process.system('pgrep khugepaged', shell=True,
+                                             verbose=False)
```

Comment:
The while loop doesn't make sense to me.  E.g. The file_name might be /sys/kernel/mm/transparent_hugepage/khugepaged/scan_sleep_millisecs, which won't change the status of khugepaged. If khugepaged is alive,  we always get CmdError in L301,  ret is from L288 other than L296 and it's always 0 since L331.

Reply:
you are right and I have made the changes as suggested.